apps/src/code-studio/pd/regional_partner_mini_contact/RegionalPartnerMiniContact.jsx

```diff
@@ -139,3 +167,101 @@ export default class RegionalPartnerMiniContact extends React.Component {
     }
   }
 }
+
+export class RegionalPartnerMiniContactPopupLink extends React.Component {
+  static propTypes = {
+    zip: PropTypes.string,
+    notes: PropTypes.string,
+    sourcePageId: PropTypes.string.isRequired,
+    children: PropTypes.node.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showing: false,
+      options: null
+    };
+
+    $.ajax({
+      type: 'GET',
+      url: '/dashboardapi/v1/users/me/contact_details'
+    })
+      .done(results => {
+        this.setState({
+          options: {
+            user_name: results.user_name,
+            email: results.email,
+            zip: this.props.zip || results.zip,
+            notes: this.props.notes || results.notes
+          }
+        });
+      })
+      .fail(() => {
+        this.setState({
+          options: {zip: this.props.zip, notes: this.props.notes}
+        });
+      });
+  }
+
+  open = () => {
+    this.setState({showing: true});
+  };
+
+  close = () => {
+    this.setState({showing: false});
+  };
+
+  render() {
+    return (
+      <span>
+        <span onClick={this.open}>{this.props.children}</span>
+        {this.state.showing && (
+          <div
```

Comment:
Just curious - is there a reason why aren't using an existing component like the react bootstrap modal? I'd rather leverage our existing components, unless there is something different about this modal that is incompatible.

Reply:
Good point.  I've used the existing react modal instead, though it still needed a little extra styling to look okay on pegasus.